internal/apptoken/app_token_grant_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package apptoken
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/apptoken/store"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestNewAppTokenGrant(t *testing.T) {
+	t.Parallel()
+	textCtx := context.Background()
+	const (
+		testAppTokenId = "app-token-id"
+		testGrant      = "id=*;type=*;actions=*"
+	)
+
+	tests := []struct {
+		name            string
+		appTokenId      string
+		grant           string
+		want            *AppTokenGrant
+		wantErrContains string
+		wantErrMatch    *errors.Template
+	}{
+		{
+			name:       "success",
+			appTokenId: testAppTokenId,
+			grant:      testGrant,
+			want: &AppTokenGrant{
+				AppTokenGrant: &store.AppTokenGrant{
+					AppTokenId:     testAppTokenId,
+					RawGrant:       testGrant,
+					CanonicalGrant: testGrant,
+				},
+			},
+		},
+		{
+			name:            "missing app token id",
+			grant:           testGrant,
+			wantErrContains: "missing app token id",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
+		{
+			name:            "missing grant",
+			appTokenId:      testAppTokenId,
+			wantErrContains: "missing app token id",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
+		{
+			name:            "invalid grant",
+			appTokenId:      testAppTokenId,
+			grant:           "id=*;type=*;actions=",
+			wantErrContains: "parsing grant string",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
```

Comment:
What should be the correct error to match this against? It is currently passing but I don't think it should be.

Reply:
Fixed.